services/gateway/gateway-service.mjs

```diff
@@ -235,57 +236,28 @@ class GatewayService {
 
   /**
    * Callback for the playerUpdate message from the client.
+   * Performs some check on the update message.
    * @param {Socket} socket - The socket from which this is sent.
-   * @param {PlayerSyncMessage} origMsg - The update message.
+   * @param {PlayerSyncMessage} updateMsg - The update message.
    */
-  async onPlayerUpdate(socket, origMsg) {
-    if (socket.playerID !== origMsg.playerID) {
-      console.error(`Player '${origMsg.playerID}' tries to update player '${socket.playerID}'s data, which is invalid.`);
+  async onPlayerUpdate(socket, updateMsg) {
+    if (socket.playerID !== updateMsg.playerID) {
+      console.error(`Player '${updateMsg.playerID}' tries to update player '${socket.playerID}'s data, which is invalid.`);
       return;
     }
 
-    const msg = PlayerSyncMessage.fromObject(socket.playerData);
-    msg.facing = origMsg.facing;
```

Comment:
I think it makes sense to explicitly copy over the fields that we want to allow the users to update (the way it's done in the deleted code). This way, whenever a new field is added, there's lesser chance for us to introduce logical bug that allows user to manipulate the field without check.

Reply:
One of the purposes to use `PlayerSyncMessage` class is to restrict the fields that can be updated. I think it is unnecessary for the server to construct the update message again.